packages/flutter/test/material/dialog_test.dart

```diff
@@ -111,6 +111,28 @@ void main() {
     expect(materialWidget.color, customColor);
   });
 
+  testWidgets('dialogBackground defaults to ColorScheme.surface', (WidgetTester tester) async {
+    const Dialog dialog = Dialog(
```

Comment:
```suggestion
    final ThemeData theme = ThemeData(colorScheme: ThemeData().colorScheme.copyWith(
      surface: Colors.orange,
      background: Colors.green,
    ));
    const Dialog dialog = Dialog(
```

Theme variable can be passed to the `MaterialApp` and Dialog background can be tested against `surface` from the `theme` variable.